articles/compliance/gdpr/features-aiding-compliance/user-consent.md

```diff
@@ -0,0 +1,353 @@
+        "text": "{\"user_metadata\":{\"consent\": {\"given\":true, \"date\":\"01/23/2018\", \"text_details\":\"some-url\"}}}"
+    }
+}
+```
+
+This will add a new property to the user profile, the **user_metadata.consent.date**, which will hold the date the customer consented. The response will be the full user profile. The updated metadata will look like this:
+
+```json
+{
+  "consent": {
+    "given": true,
+    "date": "01/23/2018",
+    "text_details": "some-url"
+  }
+}
+```
+
+### Export consent information
+
+To export a list of your users using the Management API, use [the User export endpoint](/users/search#user-export). 
+
+This endpoint creates a job that exports all users associated with a connection. You will need the Id of the connection. To find this Id, use [the Get Connections endpoint](/api/management/v2#!/Connections/get_connections) (you can set the **name** parameter to the name of the connection to retrieve only this one).
+
+Once you have the connection Id and a [Management API token](/api/management/v2/tokens), you are ready to start exporting users. For a sample request and response see [User Export](/users/search#user-export).
+
+---
+
+:::panel What else do I have to do?
+- Determine how you want to track consent. We recommend putting in not just the date but the correct version of terms and conditions, specifying the version, and including an array for those who will withdraw, provide and withdraw consent again.
+- Choose where you want to store consent, in Auth0's database or elsewhere
+:::
+
+## Withdraw consent
+
+The user should have the option to withdraw consent using your app. This option should be easily accessible, and clearly distinguishable. Once the users decides to withdraw their consent, you should take action. First, you have to decide how you will handle withdrawal of consent: will you delete the users or flag them as deleted?
+
+### Delete user
+
+To delete a user, use the [Delete a user endpoint](/api/management/v2#!/Users/delete_users_by_id). 
+
+```har
+{
+    "method": "DELETE",
+    "url": "https://${account.namespace}/api/v2/users/USER_ID",
+    "httpVersion": "HTTP/1.1",
+    "headers": [{
+        "name": "Authorization",
+        "value": "Bearer YOUR_MGMT_API_ACCESS_TOKEN"
+    }]
+}
+```
+
+The response body for this endpoint is empty, so if you want to confirm that the user was successfully deleted try to [retrieve the user using their email](/users/search#users-by-email). If the endpoint returns an error, then your call to delete the user was successful.
+
+### Flag user as deleted
+
+If you don't want to completely delete the user, flag their profile as deleted at the `app_metadata` (endpoint: [Update a user](/api/management/v2#!/Users/patch_users_by_id)). Then, add some code that will make the authentication process to fail for any user with their profile flagged as such. This way you can keep a record of deleted users in case you need to refer to this information in the future.
```

Comment:
If you don't want to delete the user, you can flag their profile as deleted using the **app_metadata** field using the [Update a user](/api/management/v2#!/Users/patch_users_by_id) endpoint.

You can then configure the authentication process to fail for users who have such a flag in their profile.

This allows you to keep a record of deleted users for future use.